toolchain/check/generic.cpp

```diff
@@ -373,26 +402,34 @@ auto FinishGenericDefinition(Context& context, SemIR::GenericId generic_id)
   context.generic_region_stack().Pop();
 }
 
-auto MakeSpecific(Context& context, SemIRLoc loc, SemIR::GenericId generic_id,
-                  SemIR::InstBlockId args_id) -> SemIR::SpecificId {
-  auto specific_id = context.specifics().GetOrAdd(generic_id, args_id);
-
+static auto ResolveSpecificDeclaration(Context& context, SemIRLoc loc,
+                                       SemIR::SpecificId specific_id) -> void {
   // If this is the first time we've formed this specific, evaluate its decl
   // block to form information about the specific.
   if (!context.specifics().Get(specific_id).decl_block_id.is_valid()) {
+    // Put a placeholder value in the decl block so we won't attempt to
+    // recursively resolve the same specific.
```

Comment:
Can you also add a TODO comment to ResolveSpecificDefinition to handle recursive resolution of the same generic definition?

Reply:
Done.